cargo-pgx/src/command/init.rs

```diff
@@ -166,10 +169,18 @@ pub(crate) fn init_pgx(pgx: &Pgx) -> eyre::Result<()> {
     for pg_config in output_configs.iter() {
         validate_pg_config(pg_config)?;
 
-        let datadir = pg_config.data_dir()?;
-        let bindir = pg_config.bin_dir()?;
-        if !datadir.exists() {
-            initdb(&bindir, &datadir)?;
+        if no_init_db {
+            println!(
+                "{} initdb as --no-init-db is specified {}",
+                "   Skipping ".bold().green(),
+                no_init_db.bold().green(),
+            );
+        } else {
+            let datadir = pg_config.data_dir()?;
+            let bindir = pg_config.bin_dir()?;
+            if !datadir.exists() {
+                initdb(&bindir, &datadir)?;
+            }
```

Comment:
Currently we avoid running `initdb` if `pg_config.data_dir()` exists.

But presumably, in the case that the user has specified a given path, the database should in fact _always_ not be initialized by us, as the user wants a specific database to be used and likely has it configured the way they like it, or will want to do so themselves. It may be that we will not be able to execute successive commands using `cargo pgx` if `initdb` has not been run, but we will run into that issue if this flag is passed as well, so we must handle that case anyways. After all, `cargo pgx` cannot simply assume that the system state has remain unchanged between runtime sessions. This is especially true if `cargo pgx` is pointed at a specific PostgreSQL database and so is not "in control" of it.

Is this hypothesis incorrect?

Reply:
This hypothesis **is** incorrect! PGX expects a specific path for the `data_dir` to exist, it's not actually controlled by `pg_config`, see: https://github.com/tcdi/pgx/blob/master/pgx-utils/src/pg_config.rs#L199-L202

Reply:
`datadir` is currently hardcoded to `data-<major-version>`, user is not allow to specify this.